python/pfnet/cbus.pyx

```diff
@@ -577,13 +577,41 @@ cdef class Bus:
             else:
                 return np.array(r)
 
-    property v_max:
-        """ Bus volatge upper bound (p.u. bus base kv) (float). """
-        def __get__(self): return cbus.BUS_get_v_max(self._c_ptr)
+    property v_max_reg:
+        """ Bus maximum regulated voltage magnitude (p.u. bus base kv) (float). """
+        def __get__(self): return cbus.BUS_get_v_max_reg(self._c_ptr)
 
-    property v_min:
-        """ Bus voltage lower bound (p.u. bus base kv) (float). """
-        def __get__(self): return cbus.BUS_get_v_min(self._c_ptr)
+    property v_min_reg:
+        """ Bus minimum regulated voltage magnitude (p.u. bus base kv) (float). """
+        def __get__(self): return cbus.BUS_get_v_min_reg(self._c_ptr)
+
+    property v_max_norm:
+        """ Bus normal maximum voltage magnitude violation limit (p.u. bus base kv) (float). """
+        def __get__(self):
+            return cbus.BUS_get_v_max_norm(self._c_ptr)
+        def __set__(self,value):
+            cbus.BUS_set_v_max_norm(self._c_ptr,value)
+
+    property v_min_norm:
+        """ Bus normal minimum voltage magnitude violation limit (p.u. bus base kv) (float). """
+        def __get__(self):
+            return cbus.BUS_get_v_min_norm(self._c_ptr)
+        def __set__(self,value):
+            cbus.BUS_set_v_min_norm(self._c_ptr,value)
+
+    property v_max_emer:
+        """ Bus emergency maximum voltage magnitude violation limit (p.u. bus base kv) (float). """
+        def __get__(self):
+            return cbus.BUS_get_v_max_emer(self._c_ptr)
+        def __set__(self,value):
+            cbus.BUS_set_v_max_emer(self._c_ptr,value)
+
+    property v_min_emer:
+        """ Bus emergency minimum voltage magnitude violation limit (p.u. bus base kv) (float). """
+        def __get__(self):
+            return cbus.BUS_get_v_min_emer(self._c_ptr)
+        def __set__(self,value):
+            cbus.BUS_set_v_min_emer(self._c_ptr,value)
```

Comment:
Docstrings are inconsistent between emer,norm and reg.  Please change to
"Bus normal/emergency/regulation maximum/minimum voltage magnitude". Do you agree?
Also, I thought we agreed on having v_max and v_min be aliases of v_max_norm, v_min_norm. Did you change your mind?